search/query.go

```diff
@@ -226,31 +226,34 @@ func addMustNotSeries(q Query) *elastic.BoolQuery {
 // filterOutCUSources - optional list of sources for which we want to filter out the CU's that connected to those sources
 //	(in order to avoid duplication between carousel and regular results).
 // titlesOnly - limit our search only to title fields: title, full_title and description in case we search for intent sources. Used for intent search.
-func createResultsQuery(resultTypes []string, q Query, docIds []string, filterOutCUSources []string, titlesOnly bool) (elastic.Query, error) {
+func createResultsQuery(options CreateResultQueryOptions) (elastic.Query, error) {
```

Comment:
Why not use SearchRequestOptions type as param?
If you prefer to create a new type then consider to use inheritance:
```
type CreateResultQueryOptions struct {
SearchRequestOptions 
}
```

Reply:
rolled back editing